systers_portal/meetup/views.py

```diff
@@ -1489,3 +1508,71 @@ def post(self, request):
         meetups = Meetup.objects.filter(date__gte=date).order_by('date')
         apiforvmsview = ApiForVmsView()
         return(apiforvmsview.return_meetup_data(meetups))
+
+
+class UpcomingMeetupsSearchView(ListView):
+    """Search Upcoming Meetups By Meetup Location, Date & Keyword and Filter by Date &
+    Distance"""
+    template_name = "meetup/list_meetup.html"
+    model = Meetup
+
+    @csrf_exempt
+    def post(self, request):
+        if request.method == 'POST':
+            date = request.POST.get('date')
+            meetup_location = request.POST.get('meetup_location')
+            keyword = request.POST.get('keyword')
+            selected_filter = request.POST.get('filter')
+            location = request.POST.get('location')
+            if meetup_location == 'Meetup Location':
+                meetup_location = ''
+            if date and meetup_location and keyword:
+                searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
+                                                         Q(title__contains=keyword), date=date,
+                                                         meetup_location__name=meetup_location)
+            elif date and meetup_location:
+                searched_meetups = Meetup.objects.filter(date=date,
+                                                         meetup_location__name=meetup_location)
+            elif date and keyword:
+                searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
+                                                         Q(title__contains=keyword), date=date,
+                                                         description__contains=keyword)
+            elif meetup_location and keyword:
+                searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
+                                                         Q(title__contains=keyword),
+                                                         meetup_location__name=meetup_location)
+            elif date:
+                searched_meetups = Meetup.objects.filter(date=date)
+            elif meetup_location:
+                searched_meetups = Meetup.objects.filter(meetup_location__name=meetup_location)
+            elif keyword:
+                searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
+                                                         Q(title__contains=keyword))
+            else:
+                searched_meetups = Meetup.objects.filter(date__gte=datetime.date.today())
+
+            results = list()
+            unit = ''
```

Comment:
Initialize unit and distance with None

Reply:
Doing this would display an invalid message like "None None" below searched meetups

Reply:
In that case, initialize distance with integer 0 instead of empty string